### ISSUE TITLE
Add index.d.ts

### ISSUE THREAD

**Issue by Mkots** [4teamwork/cypress-drag-drop#17]
For TS users it will be useful. Now, if you use Cypress with TS you'll get this error:
```ts
Property 'drag' does not exist on type 'Chainable<JQuery<HTMLAnchorElement>>'
```

**Comment by bierik** [comment]
@Mkots As typescript is a superset of javascript it should be no problem to use the plugin with cypress. Can you give me more information about how you integrated the plugin in your existing project.

**Comment by Mkots** [comment]
Yeah, I've just rewritten it to TS 
``` typescript
export class DragSimulator {
    private MAX_TRIES = 5;
    private DELAY_INTERVAL_MS = 10;
    private dataTransfer = new DataTransfer();
    private counter = 0;
    private source: HTMLElement;
    private target: HTMLElement;
    private initialSourcePosition: ClientRect;
    private position: string;

    rectsEqual(r1: ClientRect, r2: ClientRect): boolean {
        return r1.top === r2.top && 
                  r1.right === r2.right && 
                  r1.bottom === r2.bottom && 
                  r1.left === r2.left;
    }

    get dropped(): boolean {
        const currentSourcePosition = this.source.getBoundingClientRect();
        return !this.rectsEqual(this.initialSourcePosition, currentSourcePosition);
    }
    get hasTriesLeft(): boolean {
        return this.counter < this.MAX_TRIES;
    }
    private dragstart(): void {
        cy.wrap(this.source, { log: false })
            .trigger('mousedown', { which: 1, button: 0, log: false })
            .trigger('dragstart', { dataTransfer: this.dataTransfer, log: false });
    }
    private drop(): Cypress.Chainable {
        return cy
            .wrap(this.target, { log: false })
            .trigger('drop', { dataTransfer: this.dataTransfer, force: true, log: false })
            .trigger('mouseup', { which: 1, button: 0, log: false });
    }
    private dragover(): Cypress.Chainable | boolean {
        if (!this.dropped && this.hasTriesLeft) {
            this.counter += 1;
            return cy
                .wrap(this.target, { log: false })
                .trigger('dragover', { dataTransfer: this.dataTransfer, position: this.position, log: false })
                .wait(this.DELAY_INTERVAL_MS, { log: false })
                .then(() => this.dragover());
        }
        if (!this.dropped) {
            return this.drop().then(() => {
                console.error(`Exceeded maximum tries of: ${this.MAX_TRIES}, aborting`);
                return false;
            });
        }
        return this.drop().then(() => true);
    }
    private init(source: HTMLElement, target: HTMLElement, position: string): Cypress.Chainable {
        this.source = source;
        this.target = target;
        this.position = position;
        this.counter = 0;

        this.dragstart();

        return cy.wait(this.DELAY_INTERVAL_MS, {log: false}).then(() => {
            this.initialSourcePosition = this.source.getBoundingClientRect();
            return this.dragover();
        });
    }
    simulate(...args: any[]): Cypress.Chainable {
        const sourceWrapper: JQuery<HTMLElement> = args[0];
        const targetSelector: string = args[1];
        const position: string = args[2] ? args[2] : 'top';
        return cy
            .get(targetSelector)
            .then(targetWrapper => this.init(sourceWrapper.get(0), targetWrapper.get(0), position));
    }
}
```

**Comment by bierik** [comment]
@Mkots I do not think you have to rewrite the plugin in TS. Simply install the plugin and import it as described in the README. This should work. Maybe you have to disable the typecheck of TS for the plugin.

**Comment by smart625** [comment]
@bierik For TS I met the same issue, even though I have import the plugin in TS following RWADME.  So can you take a look at it or this is an only issue for TS users? 
![image](https://user-images.githubusercontent.com/24734147/71651033-8aa8a680-2d55-11ea-9c0a-1eb2290356db.png)


Another issue is when I use this plugin to drag and drop element

**Comment by bierik** [comment]
@smart625 @Mkots I never used TS so no idea what to do exactly. Should I do something like this: https://github.com/abramenal/cypress-file-upload/blob/master/index.d.ts to get rid of the error?
@smart625 About the `other Issues`: Right the single `center` argument does not work anymore. The `.drag` command will accept an options object in the next version. This is already fixed on master. Sorry for the confusion.

**Comment by ishanshukla97** [comment]
I added this to my cypress/support/index.d.ts

`/// <reference types="cypress" />

declare namespace Cypress {
    interface Chainable {
      /**
       * Custom command to select DOM element by data-cy attribute.
       * @example cy.dataCy('greeting')
      */
      drag(value: string): Chainable<Element>
    }
  }`

**Comment by bierik** [comment]
Fixed in Version 1.6.0 by https://github.com/4teamwork/cypress-drag-drop/pull/54
